tests/test_backups.py

```diff
@@ -0,0 +1,171 @@
+        image="t4c/client/backup", environment=env, network="host"
+    ) as container:
+        yield container
+
+
+@pytest.fixture(name="t4c_server_container")
+def fixture_t4c_server_container(request) -> containers.Container:
+    env: dict[str, str] = {"REST_API_PASSWORD": "password"} | request.param
+
+    ports: dict[str, int] = {
+        "2036/tcp": 2036,
+        "8080/tcp": 8081,
+        "12036/tcp": 12036,
+    }
+
+    with conftest.get_container(
+        image="t4c/server/server",
+        ports=ports,
+        environment=env,
+        path=(pathlib.Path(__file__).parents[0])
+        / "t4c-server-test-data"
+        / "data"
+        / CAPELLA_VERSION,
+        mount_path="/data",
+    ) as container:
+        conftest.wait_for_container(
+            container, "!MESSAGE Warmup done for repository test-repo."
+        )
+        yield container
+
+
+@pytest.mark.t4c_server
+@pytest.mark.parametrize(
+    "t4c_server_container,t4c_backup_container",
+    [
+        pytest.param({"ENABLE_CDO": "true"}, {"CONNECTION_TYPE": "telnet"}),
+        pytest.param(
+            {"ENABLE_CDO": "false"},
+            {"CONNECTION_TYPE": "telnet"},
+            marks=pytest.mark.skipif(
+                condition=is_capella_6_x_x(),
+                reason="CDO disabled by default for capella >= 6.0.0",
+            ),
+        ),
+        pytest.param(
+            {"ENABLE_CDO": "false"},
+            {
+                "CONNECTION_TYPE": "http",
+                "HTTP_LOGIN": "admin",
+                "HTTP_PASSWORD": "password",
+                "HTTP_PORT": "8081",
+            },
+            marks=pytest.mark.skipif(
+                condition=is_capella_5_x_x(),
+                reason="conncetion type http is not supported for capella < 6.0.0",
+            ),
+        ),
+    ],
+    indirect=True,
+)
+def test_model_backup_happy(
+    local_git_container,  # pylint: disable=unused-argument
+    t4c_server_container,  # pylint: disable=unused-argument
+    t4c_backup_container,
+    tmp_path: pathlib.Path,
+):
+    conftest.wait_for_container(t4c_backup_container, "Backup finished")
+
+    subprocess.run(
+        [
+            "git",
+            "clone",
+            "http://localhost:9010/git/git-test-repo.git",
+            tmp_path,
+        ],
+        check=True,
+    )
+    try:
+        subprocess.run(
+            ["git", "switch", "backup-test"], check=True, cwd=tmp_path
+        )
+    except subprocess.CalledProcessError:
+        log.debug("backup failed - backup-test branch does not exists")
+        raise
+
+    assert os.path.exists(tmp_path / ".project")
+    assert os.path.exists(tmp_path / "test-project.afm")
+    assert os.path.exists(tmp_path / "test-project.aird")
+    assert os.path.exists(tmp_path / "test-project.capella")
+
+
+@pytest.mark.t4c_server
+@pytest.mark.parametrize(
+    "t4c_server_container,t4c_backup_container",
+    [
+        ({"ENABLE_CDO": "true"}, {"CONNECTION_TYPE": "unknown"}),
```

Comment:
```suggestion
        # CONNECTION_TYPE "unknown" is not allowed
        ({"ENABLE_CDO": "true"}, {"CONNECTION_TYPE": "unknown"}),
```

For each test case, I'd give a short explanation why it should fail.